### DISCUSSION TITLE
Portainer, encryption key env variable, after the fact?

### DISCUSSION THREAD

**Discussion by pcm1ke** [teslamate-org/teslamate#3090] [Q&A]
I’ve been running Teslamate for well over a year now and some time ago I switched to managing all my containers via stacks within portainer. Obviously this makes this easier to update, but it also leads to not always reading release notes before updating things. Case in point, with the upgrade to 1.27 the new encryption key environment variable was introduced and I’m literally just finding out now when I’m looking to upgrade Postgres to the next major version at some point. Thankfully, my installation never stopped working and continues to work, however I have some questions surrounding how to move forward from here.

1) The release notes indicate that if the encryption key environment variable is NOT set in the YML that a random one will be assigned and the logs will contain further instructions. I updated a long time ago without even knowing about this, so is there any chance I’d knowing what those further instructions were?

*2) Since I upgraded long ago and a random encryption key was obviously generated, is it now safe to introduce the encryption key environment variable of my choice into my YML or will this break things?

Bonus question, will the release notes indicate when we should move to the next major version of Postgres? Obviously portainer keeps indicating that my Postgres image is out of date, but that’s because my stack is referencing the older version and the image will never update beyond that. I don’t want to jump the gun and update Postgres and break teslamate, so just looking for guidance.

Thank you 🙏🏻

**Comment by cwanja** [comment]
1. If you restart the container without an encryption key in the YML, I believe it should produce a new encryption key in the logs and you are free to begin using that. 

2. Totally fine. In fact, baring you do not mind a warning on the Home Screen, I do not think there's a requirement to use a manually inserted encryption key. You could just keep using the one generated by the application at runtime and move along. Obviously this has risks if your installation is public facing.

3. I feel like I recall seeing in a release notes when the application moved from 11 to 12 and or 12 to 13. I think the default Docker YML now says 14, so you could in theory upgrade to 14.

**Reply by pcm1ke** [reply]
@cwanja Thank you!

Upon further investigation, I guess I did set my own encryption key at some point, so my fears were unfounded. Also, I guess Postgres was recently updated to 15 on docker hub, so that’s why portainer is always reporting it as out of date. I’ll just ignore that for now and wait for guidance in release notes hopefully. 👍🏻
